File: PTM/webapp/node.py
class Node:
    def __init__(self,name,la,mi):
        self.label=str(name)
        self.failureRate=la
        self.repairRate=mi
    
    def __repr__(self):
        return self.label

    
def node_from_dict(d):
    name=str(d["label"])
    la=d["failureRate"]
    mi=d["repairRate"]
    return Node(name,la,mi)

    

File: PTM/client/models/node.ts
import { FailRepairRate } from './rates';

export class Node extends FailRepairRate {
    private label: string;
    private id: string; 

    constructor(label:string, id:string, failureRate:number, repairRate:number) { 
        super();
        this.label = label;
        this.id = id;
        super.setFailureRate(failureRate);
        super.setRepairRate(repairRate); 
     }

    getLabel(): string {
        return this.label;
    }

    setLabel(label: string): void {
        this.label = label;
    }

    getId(): string {
        return this.id;
    }

    setId(id: string): void {
        this.id = id;
    }
    
}